File: fe/src/app/order/order.ts
import {Session} from "../session/session";
import {User} from "../user/user";
/**
 * Created by bondarm on 07.01.17.
 */
export class Order {
    id: number;
    date: Date;
    user: User;
    session: Session;
    regTicketsCount: number;
    vipTicketsCount: number;
    price: number;
    discounts: string[];
    confirmed: boolean;

    constructor(
        session: Session,
        user: User
    ) {
        this.id = -1;
        this.date = new Date;
        this.user = user;
        this.session = session;
        this.regTicketsCount = 0;
        this.vipTicketsCount = 0;
        this.price = 0;
        this.discounts = [];
        this.confirmed = false;

        console.log('order' + JSON.stringify(this));
    }

}

File: src/main/java/com/bondarmih/ticketbooking/data/entity/Order.java
package com.bondarmih.ticketbooking.data.entity;

import org.hibernate.annotations.Type;
import org.joda.time.DateTime;

import javax.persistence.*;
import java.util.Date;
import java.util.Set;

/**
 * Created by bondarm on 22.12.16.
 */

@Entity
@Table(name = "Order")
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "id")
    private long id;

    @ManyToOne
    private User user;

    @Column(name = "date")
    private Date date;

    @ManyToOne
    private MovieSession movieSession;

    @Column(name = "regSeats")
    private int regSeats;

    @Column(name = "vipSeats")
    private int vipSeats;

    @Column(name = "price")
    private double price;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public int getRegSeats() {
        return regSeats;
    }

    public void setRegSeats(int regSeats) {
        this.regSeats = regSeats;
    }

    public int getVipSeats() {
        return vipSeats;
    }

    public void setVipSeats(int vipSeats) {
        this.vipSeats = vipSeats;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public MovieSession getMovieSession() {
        return movieSession;
    }

    public void setMovieSession(MovieSession movieSession) {
        this.movieSession = movieSession;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }
}
